segfast/segyio_loader.py

```diff
@@ -125,32 +122,40 @@ def load_headers(self, headers, reconstruct_tsf=True, sort_columns=True, tracewi
             If str, then type of progress bar to display: `'t'` for textual, `'n'` for widget.
         """
         _ = kwargs
-        headers = list(headers)
+        headers = self.make_headers(headers)
```

Comment:
I'd check that all `headers` are "standard" after this line: now you'll silently load a header if has a custom dtype, but its starting byte matches that of a standard header.
`is_standard` may be a property of `TraceHeader` class.